### PR TITLE
Replace link previews with new unfurled data from the message

### PR DESCRIPTION
**Pull request by igor-sirotin**
Fixes https://github.com/status-im/status-desktop/issues/11474

### What does the PR do

Show link previews from message-attached unfurled `linkPreviews`.

### Affected areas

Chat

### Screenshot of functionality (including design for comparison)

Screenshot 2023-07-20 at 14 30 43

I should mention that with this update image, gif and Status links will not have a preview anymore. 
Minor updates should be done in desktop to support this after implementing:
- https://github.com/status-im/status-go/issues/3761
- https://github.com/status-im/status-go/issues/3762

Screenshot 2023-07-20 at 14 30 57

### CODE REVIEW

**Comment by igor-sirotin** [comment]
@jrainville alghouth Image, GIF and Status links are not unfurled anymore, I think it's fine to merge it and make left changes in 1-2 weeks when status-go is ready. 
wdyt?

**Comment by jrainville** [comment]
> @jrainville alghouth Image, GIF and Status links are not unfurled anymore, I think it's fine to merge it and make left changes in 1-2 weeks when status-go is ready. wdyt?

Images and gifs is fine. Not having Status links can be annoying to join test communities though. Is there a way to still have them for now?

**Comment by igor-sirotin** [comment]
> > @jrainville alghouth Image, GIF and Status links are not unfurled anymore, I think it's fine to merge it and make left changes in 1-2 weeks when status-go is ready. wdyt?
> 
> Images and gifs is fine. Not having Status links can be annoying to join test communities though. Is there a way to still have them for now?

hmm. I though that we could make the URL opening, just not preview will be shown.
And actually that already works! 🙂 

https://github.com/status-im/status-desktop/assets/25482501/37f103c7-5966-4cd3-b1e9-71cc5dfc4908

File: src/app/modules/shared_models/message_item.nim [hunk 1]
```diff
@@ -87,6 +90,7 @@ proc initItem*(
     sticker: string,
     stickerPack: int,
     links: seq[string],
```
Comment (jrainville) [inline src/app/modules/shared_models/message_item.nim:92]: Is `links` still used?

**Review by jrainville — APPROVED** [review]
LGTM

File: ui/imports/shared/views/chat/LinksMessageView.qml [hunk 2]
```diff
@@ -13,122 +13,81 @@ import shared.panels 1.0
...
-                    },
-                    State { 
-                        name: "loadImage"
-                        when: unfurl && isImage
-                        PropertyChanges { target: linkMessageLoader; sourceComponent: unfurledImageComponent }
-                    },
-                    State { 
                         name: "loadLinkPreview"
-                        when: unfurl && !isImage && !isStatusDeepLink
+                        when: !linkMessageLoader.isImage && !linkMessageLoader.isStatusDeepLink
```
Comment (alexjba) [inline ui/imports/shared/views/chat/LinksMessageView.qml:54]: Missing properties
Do we plan to extend the model with these roles?

**Review by alexjba — APPROVED** [review]
LTGM! Glad to see this change 👍 

Minor comment below

File: ui/imports/shared/views/chat/LinksMessageView.qml [hunk 3]
```diff
@@ -13,122 +13,81 @@ import shared.panels 1.0
 import shared.stores 1.0
 import shared.controls.chat 1.0
 
-Column {
+ColumnLayout {
     id: root
 
     property var store
     property var messageStore
+    property var linkPreviewModel
```
[the remark this answers is not in the document]
Comment (igor-sirotin) [reply ui/imports/shared/views/chat/LinksMessageView.qml:21]: well it should be just `LinkPreviewModel` class instance 😄 
I also listed the roles below as `required` properties:
https://github.com/status-im/status-desktop/blob/5feff517180f590a1e54b35f964ef412b230c25b/ui/imports/shared/views/chat/LinksMessageView.qml#L36-L44

File: src/app/modules/shared_models/message_item.nim (hunk 1 above)
Comment (igor-sirotin) [reply src/app/modules/shared_models/message_item.nim:92]: Well, it's still part of the Message in status-go. Don't now if it makes sense to drop it now.

File: ui/imports/shared/views/chat/LinksMessageView.qml (hunk 2 above)
Comment (igor-sirotin) [reply ui/imports/shared/views/chat/LinksMessageView.qml:52]: https://github.com/status-im/status-desktop/issues/11634

**Pull request merged**